### ISSUE TITLE
v1.0.0-alpha.2 - Features

### ISSUE THREAD

**Issue by Nate-Wilkins** [Flet/eslint-to-esformatter#5]
## `v1.0.0-alpha.2`

Working off of eslint `0.19.0`
#### Framework
- [x] #3 logging
- [x] #4 deprecated rules
- [x] rule documentation (things not supported in eslint/esformatter)
#### Core Rules
- [x] `brace-style`
  - [x] stroustrup
- [x] `indent`
  - [x] indentSwitchCase
- [x] `comma-spacing`
- [x] `eol-last`
- [x] `newline-after-var`
- [x] `key-spacing`
- [ ] `generator-star-spacing`
- [ ] `no-multi-spaces`
- [ ] `no-multiple-empty-lines`
- [ ] `no-spaced-func`
- [ ] `no-trailing-spaces`
- [ ] `padded-blocks`
- [ ] `semi-spacing`
- [ ] `space-after-keywords` @Nate-Wilkins
- [ ] `space-before-blocks` @Nate-Wilkins
- [ ] `space-before-function-paren`
- [ ] `space-in-brackets`
- [ ] `space-in-parens`
- [ ] `space-infix-ops` @Nate-Wilkins
- [ ] `space-return-throw-case`
- [ ] `space-unary-ops`
- [ ] `spaced-line-comment`
### Other
- [x] All deprecated rules
- [x] All informative rules
- [x] All unsupported rules (requires esformatter plugin(s))

**Comment by Nate-Wilkins** [comment]
I have code for the rules listed (block-scoped-var, brace-style, indent, coma-spacing, eol-last, newline-after-var)

@Flet Add anything you want here!

**Comment by Flet** [comment]
`key-spacing` and `indent` are done and tested... added `key-spacing` to the list.

**Comment by Nate-Wilkins** [comment]
Pushed up `indentSwitchCase` and a few other things with comments in commits.

**Comment by Flet** [comment]
You have been busy! :+1:

**Comment by Nate-Wilkins** [comment]
Yup :+1: as a side note I can't get `faucet` to work... I'm using windows 8 which looks to be the culprit or my node version I'll open an issue on `faucet`'s repo if I'm still having issues

Adding the `debug` module now along with the rest of the rules I have.

**Comment by Flet** [comment]
Ah, `faucet` will only execute tests that are at the root of the `test` directory. I just rearranged things in this commit https://github.com/Flet/eslint-to-esformatter/commit/abea90a9e05dbdcffe12233148aad9fcb05d1bef to make it work.
- put rules at the root
- moved  `test-rule.js` into new dir `lib`

Also I just added a little skip option to get around `blocked-scope-var` having no tests. 

So now running `faucet` does this for me:

``` bash
⨯ brace-style
  not ok 3 format using: rule on, stroustrup
    ---
      operator: equal
      expected:
        'function foo() {\n  return true;\n}\n\nif (foo) {\n  bar();\n}\n\ntry {\n  somethingRisky();\n}\ncatch (e) {\n  handleError();\n}\n\nif (foo) {\n  bar();\n}\nelse {\n  baz();\n}'
      actual:
        'function foo() {\n  return true;\n}\n\nif (foo) {\n  bar();\n}\n\ntry {\n  somethingRisky();\n} catch (e) {\n  handleError();\n}\n\nif (foo) {\n  bar();\n} else {\n  baz();\n}'
      at: Test.<anonymous> (/home/icmpdev/code/eslint-to-esformatter/test/lib/test-rule.js:11:16)
    ...
✓ indent
✓ key-spacing
# tests 14
# pass  13
⨯ fail  1
```

Is it working for you with these updates?

**Comment by Nate-Wilkins** [comment]
That's expected behavior atm https://github.com/Flet/eslint-to-esformatter/commit/9fa6f7c46de2b1c3d03f952d4c077e93cea00679 where the `CatchKeyword` doesn't get the correct line break

**Comment by Flet** [comment]
Cool, does fauct work on windows for you?

**Comment by Nate-Wilkins** [comment]
No same thing unfortunately.
![capture](https://cloud.githubusercontent.com/assets/2957868/7257216/f2db0898-e821-11e4-89c6-26776bf32e0e.PNG)

**Comment by Nate-Wilkins** [comment]
Added [leasot](https://github.com/pgilad/leasot) to parse out TODO comments. Currently not in 'gulp test' or if we want it at all. Just need a way to keep track of the things I can't fix atm.

**Comment by Flet** [comment]
Will this work?
https://github.com/Flet/eslint-to-esformatter/commit/addbd71bd004b76b5163dda01a58474aa6177f65

basically just run `npm run todo` and it will scan the whole codebase and write TODO.md at root.

**Comment by Flet** [comment]
RE: `faucet`, dumping it for now.

Instead just running plain `tape` and piping to `tap-difflet` which will highlight diffs in output. Tested and working on Windows :+1: 
https://github.com/Flet/eslint-to-esformatter/commit/761259d7f6714099f97f28e60b57bd87116809df

**Comment by Nate-Wilkins** [comment]
:+1: 
![](http://blog.codinghorror.com/content/images/uploads/2007/03/6a0120a85dcdae970b0128776ff992970c-pi.png)

**Comment by Nate-Wilkins** [comment]
Updated again. Adding all the informative, deprecated, and unsupported rules now.

**Comment by Flet** [comment]
Heh, `leasot` kinda goes overboard with the TODO headings in its output!
